### DISCUSSION TITLE
How to type variables used in [bracket] notation?

### DISCUSSION THREAD

**Discussion by nkoehring** [ivanhofer/typesafe-i18n#546] [Q&A] [answered]
Related to question https://github.com/ivanhofer/typesafe-i18n/discussions/174#discussion-3876503 but I'm not sure how well you can see answers after the question is marked as answered, so here for clarity a new one:

When using bracket notation with the i18nObject, like for example `LL.foo[bar]()`, what type needs `bar` to be?

**Comment by ivanhofer** [comment, marked as answer]
@nkoehring you are probably looking for something like this:
```ts
import type { Translations } from '$i18n/i18n-types'
const key = 'something' as keyof Translations['foo']
LL.foo[key]()
```

**Reply by nkoehring** [reply]
That works, thanks!

**Comment by nkoehring** [comment]
Important to note: If namespaces are used, the type should be `Translations`, plural, instead of `Translation`, because the former includes all Namespaces, the latter explicitly forbids namespace keys and would end up as `never `:

```ts
import type { Translations } from '$i18n/i18n-types'
const key = 'something' as keyof Translations['foo']
LL.foo[key]()
```

**Reply by ivanhofer** [reply]
right, thanks for the update. I'll edit my answer above
